opcuapuffin/src/tests.rs

```diff
@@ -0,0 +1,56 @@
+use opcua::puffin::signature::{fn_hello, fn_acknowledge, fn_reverse_hello};
+use opcua::types::{AcknowledgeMessage, HelloMessage, ReverseHelloMessage, UAString};
+
+use puffin::codec::CodecP;
+
+#[test]
+pub fn client() {
+
+    let max_size = 5000;
+    let mut send_buffer: Vec<u8> = Vec::with_capacity(max_size);
+
+    let hello_message: HelloMessage = fn_hello(
+        &UAString::from("opc.tcp://PenDuick:53530/OPCUA/SimulationServer"),
```

Comment:
Two points:
1. Construct a `term` rather than a `message`. Evaluate and encode and compare as you do here.
2. To do so, make sure all arguments can be produced with a term. For example here, I don't think there is a term producing a `UAString`.